srt-tokio/src/listener/request.rs

```diff
@@ -0,0 +1,95 @@
+use std::{io::ErrorKind, net::SocketAddr, time::Instant};
+
+use bytes::Bytes;
+use futures::{
+    channel::{mpsc, oneshot},
+    SinkExt,
+};
+use srt_protocol::settings::KeySettings;
+use srt_protocol::{connection::ConnectionSettings, listener::*, packet::*};
+
+use crate::{watch, SocketStatistics, SrtSocket};
+
+pub fn new_connection_request(
```

Comment:
Any reason why this isn't in `impl ConnectionRequest`?

Reply:
I wanted to accommodate the value being held by another thread/task, potentially in a data structure of some kind, returning impl would unnecessarily force said data structure to be parametric/generic.

Reply:
I didn't mean returning impl Trait, just an associated method (inside an impl struct {}) block

Reply:
Oh, right, I'll revisit that, I don't recall why I did it this way